traits/has_traits.py

```diff
@@ -2188,6 +2188,20 @@ def class_editable_traits ( cls ):
         return names
 
     class_editable_traits = classmethod( class_editable_traits )
+    
+    def visible_traits ( self ):
+        """Returns an alphabetically sorted list of the names of non-event
+        trait attributes associated with the current object, that should be GUI visible
+        """
+        return self.trait_names( type = not_event, editable = not_false, visible = not_false )
```

Comment:
I agree it is the existing behaviour but it is based on `editable_traits`, which gives the list of editable trait. What you are adding here is the list of visible traits and replaces the usage of the old `editable_traits()` method. I think that `should be GUI visible` means editable AND non-editable traits.  If you only want the editable traits, why adding a new `visible_traits()` method?